build.py

```diff
@@ -35,8 +35,9 @@
 
 script_start_time = timer()
 
-ios_sim_device = "iPhone 14"
-ios_sim_dest = "-destination 'platform=iOS Simulator,name=" + ios_sim_device + ",OS=16.4'"
+ios_sim_device_type = "iPhone 14"
```

Comment:
This change was done so that `device_guids.get_ios(ios_sim_device_exact_name)` can get this exact SImulator. If the call will be `device_guids.get_ios(ios_sim_device)` it would get the first `iPhone 14` simulator which can be with iOS 17 depending on which way they are installed.
We needed to separate between the `type` of the device and the exact `device name` and the `destination`.
We had to do this separation because at some point when XCode 15 beta got released iPhone 14 simulators got updated to iOS 17 causing the unit tests to fail